### PR TITLE
SPM-1083: Fix package_latest_cache containing no summary

### PR DESCRIPTION
**Pull request by semtexzv**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: database_admin/schema/create_schema.sql [hunk 1]
```diff
@@ -938,7 +938,7 @@ SELECT DISTINCT ON (p.name_id) p.name_id, p.id as package_id, sum.value as summa
 FROM package p
          INNER JOIN strings sum on p.summary_hash = sum.id
          LEFT JOIN advisory_metadata am on p.advisory_id = am.id
-ORDER BY p.name_id, am.public_date DESC NULLS LAST;
+ORDER BY p.name_id, am.public_date DESC NULLS LAST, sum.value DESC NULLS LAST;
```
Comment (Josca) [inline database_admin/schema/create_schema.sql:941]: @semtexzv it doesn't work. Try this in QA:
~~~SQL
select * from (
SELECT DISTINCT ON (p.name_id) p.name_id, p.id as package_id, sum.value as summary
FROM package p
         INNER JOIN strings sum on p.summary_hash = sum.id
         LEFT JOIN advisory_metadata am on p.advisory_id = am.id
ORDER BY p.name_id, am.public_date DESC NULLS LAST, sum.value DESC NULLS LAST) t
where summary = '';
~~~
... it returns many rows.

**Review by Josca — CHANGES_REQUESTED** [review]

File: database_admin/schema/create_schema.sql (hunk 1 above)
Comment (semtexzv) [reply database_admin/schema/create_schema.sql:941]: It returns 1700 rows vs 12000 returned by the query before. It seems we have some content, which has no summaries. From my exploration it's just `.src` only packages

**Review by Josca — COMMENTED** [review]
pr, commit SPM-1083

File: database_admin/migrations/068_package_latest_fix.up.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+DROP MATERIALIZED VIEW IF EXISTS package_latest_cache;
+
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS package_latest_cache
+AS
+SELECT DISTINCT ON (p.name_id) p.name_id, p.id as package_id, sum.value as summary
+FROM package p
+         INNER JOIN strings sum on p.summary_hash = sum.id
+         LEFT JOIN advisory_metadata am on p.advisory_id = am.id
+ORDER BY p.name_id, am.public_date DESC NULLS LAST, sum.value DESC NULLS LAST;
```
Comment (Josca) [inline database_admin/migrations/068_package_latest_fix.up.sql:12]: I think we can simplify that:
~~~
ORDER BY p.name_id, am.public_date, sum.value DESC NULLS LAST;
~~~

File: database_admin/schema/create_schema.sql (hunk 1 above)
Comment (Josca) [reply database_admin/schema/create_schema.sql:941]: You're right. I've also tested that. All packages without summary are `.src`.

**Review by Josca — CHANGES_REQUESTED** [review]
One note - and pls SPM-NUM.

File: database_admin/migrations/068_package_latest_fix.up.sql (hunk 2 above)
Comment (Josca) [reply database_admin/migrations/068_package_latest_fix.up.sql:12]: Ok, it seems we can not. Ordering is specified for each column.

**Review by Josca — APPROVED** [review]

**Pull request merged**